AEP/aep-00007.md

```diff
@@ -0,0 +1,228 @@
+  ///   The message processing has been cancelled. the message can safely be removed from the queue.
+  /// </summary>
+  Cancelled,
+
+  /// <summary>
+  ///   Message has been retried too many times and is considered as poisonous for the queue
+  /// </summary>
+  Poisonous,
+}
+```
+
+Interface to insert tasks into queue is simpler.
+For a given partition, ArmoniK gives an `IEnumerable<MessageData>` where each `MessageData` represents a task.
+It contains the task identifier, the session identifier and the options of the task.
+The options has a field for the priority of the task.
+The queue system and the implementation of the interfaces are then responsible to distribute tasks.
+
+```csharp
+/// <summary>
+///   Interface to insert messages into the queue
+/// </summary>
+public interface IPushQueueStorage : IQueueStorage
+{
+  /// <summary>
+  ///   Puts messages into the queue, handles priorities of messages
+  /// </summary>
+  /// <param name="messages">Collection of messages</param>
+  /// <param name="partitionId">Id of the partition</param>
+  /// <param name="cancellationToken">Token used to cancel the execution of the method</param>
+  /// <returns>
+  ///   Task representing the asynchronous execution of the method
+  /// </returns>
+  public Task PushMessagesAsync(IEnumerable<MessageData> messages,
+                                string                   partitionId,
+                                CancellationToken        cancellationToken = default);
+}
+```
+
+```csharp
+/// <summary>
+///   Data structure to hold message data
+/// </summary>
+/// <param name="TaskId">Unique identifier of the task</param>
+/// <param name="SessionId">Unique name of the session to which this message belongs</param>
+/// <param name="Options">Task options</param>
+public record MessageData(string      TaskId,
+                          string      SessionId,
+                          TaskOptions Options);
+```
+
+## Issues with the current interface
+
+There are a few issues from the current interfaces:
+- The split of the interface into two due to the configuration. The partition should be given in the pull method directly and uniformize partition selection.
+- Clarify message processing instead of relying on the `DisposeAsync` method from the `IQueueMessageHandler`.
+
+These issues will be addressed in a new version of these interfaces.
+
+# Conclusion
+
+The architecture reflects a deliberate balance between design complexity, operational requirements, and system constraints. By combining a historical interface with strict control over message processing and lifecycle management, ArmoniK achieves a reliable and scalable system. Although the system does not adopt a fully event-driven approach, it remains robust and resilient, meeting the demands of long-running tasks and orchestration challenges. The trade-offs in this design are justified by the system’s operational stability and its ability to handle large-scale workloads efficiently.
```

Comment:
```suggestion
The architecture reflects a deliberate balance between design complexity, operational requirements, and system constraints. The interface reflects an architecture that allows a strict and precise control over message processing and lifecycle management. This allows ArmoniK to achieve a reliable and scalable system. The trade-offs in this design are justified by the system’s operational stability and its ability to handle large-scale workloads efficiently.
```

```Although the system does not adopt a fully event-driven approach, it remains robust and resilient, meeting the demands of long-running tasks and orchestration challenges.``` has been removed as it could be interpreted that a full event-driven approach should be a target. This is not the case.